superset-frontend/src/views/CRUD/data/database/DatabaseModal.tsx

```diff
@@ -115,6 +117,14 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
         ...db,
       };
 
+      // Parse 'extra' field
+      const extraParsed = JSON.parse(update?.extra || defaultExtra);
```

Comment:
I think this makes more sense being db?.extra, because that's closer to the source and matches the above.

Reply:
Good idea! Fixed in [`this commit`](https://github.com/apache/superset/pull/14293/commits/d699d3105efde7e7fa5dcd466f1c45dbfa602597)